### ISSUE TITLE
[feature request] Support defaults in template fields

### ISSUE THREAD

**Issue by dkirrane** [GoogleContainerTools/skaffold#2255]
I'm looking for something like the following. 
If `VERSION` env variable isn't set use `USERNAME` env variable otherwise use `latest` as the default for the image tag.

This is for a developer workflow where Jenkins would set the `VERSION` but the developer on their machine would user their own `USERNAME` as the image tag. And if the developer doesn't have `USERNAME` variable (like on a Mac) it will fall back to `latest`

```yaml
apiVersion: skaffold/v1beta11
kind: Config
build:
  tagPolicy:
    envTemplate:
      template: "{{.IMAGE_NAME}}:{{ .VERSION | .USERNAME | default('latest') }}"
```

**Comment by nkubala** [comment]
would it make more sense to just use different profiles here? evaluating env vars is pretty different than evaluating logical statements when generating the envTemplate tags.

```yaml
build:
  tagPolicy:
    envTemplate:
      template: "{{.IMAGE_NAME}}:{{.USERNAME}}"
profiles:
  - name: jenkins
    build:
    tagPolicy:
      envTemplate:
        template: "{{.IMAGE_NAME}}:{{.VERSION}}"
```

**Comment by no1melman** [comment]
I would prefer something like this

```yaml
build:
  artifacts:
  - image: cool
    docker:
      buildArgs:
        RUNTESTS: "{{ default .RUNTESTS '0' }}"

```

Where I can `export RUNTESTS="1"` to run the tests, or it gets defaulted to `"0"`

**Comment by gentunian** [comment]
why not to keep it "standard" or related to what helm does so it follows a common idea: https://helm.sh/docs/chart_template_guide/variables/

```
foo: {{ .MY_VAR | default "" }}
```

**Comment by okgolove** [comment]
It seems [this PR](https://github.com/GoogleContainerTools/skaffold/pull/6136) resolves the issue

**Comment by nkubala** [comment]
closed via #6136

**Comment by GregCKrause** [comment]
Could a note on this possibly be added to the [templating](https://skaffold.dev/docs/environment/templating/) guide?
